File: firefox/user/firefoxnetwork/network.js
// Copyright 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

var port = null;

var getKeys = function(obj){
   var keys = [];
   for(var key in obj){
      keys.push(key);
   }
   return keys;
}

function appendMessage(text) {
  document.getElementById('response').innerHTML += "<p>" + text + "</p>";
}

function updateUiState() {
  if (port) {
    document.getElementById('connect-button').style.display = 'none';
    document.getElementById('send-message-button').style.display = 'block';
    document.getElementById('ipaddr').style.display = 'block';
    document.getElementById('netmask').style.display = 'block';
    document.getElementById('gateway').style.display = 'block';
    document.getElementById('nameservers').style.display = 'block';
    document.getElementById('showipaddr').style.display = 'block';
    document.getElementById('shownetmask').style.display = 'block';
    document.getElementById('showgateway').style.display = 'block';
    document.getElementById('shownameservers').style.display = 'block';
    document.getElementById('showType').style.display = 'block';
    document.getElementById('ProxyType').style.display = 'block';
    document.getElementById('showproxy').style.display = 'block';
  } else {
    document.getElementById('connect-button').style.display = 'block';
    document.getElementById('send-message-button').style.display = 'none';
    document.getElementById('ipaddr').style.display = 'none';
    document.getElementById('netmask').style.display = 'none';
    document.getElementById('gateway').style.display = 'none';
    document.getElementById('nameservers').style.display = 'none';
    document.getElementById('showipaddr').style.display = 'none';
    document.getElementById('shownetmask').style.display = 'none';
    document.getElementById('showgateway').style.display = 'none';
    document.getElementById('shownameservers').style.display = 'none';
    document.getElementById('showType').style.display = 'none';
    document.getElementById('ProxyType').style.display = 'none';
    document.getElementById('showproxy').style.display = 'none';
  }
}

function manual() {
    document.getElementById('manual').checked = true;
    document.getElementById('dhcp').checked = false;
  updateUiState();
}

function dhcp() {
    document.getElementById('manual').checked = false;
    document.getElementById('dhcp').checked = true;
  updateUiState();
}

function sendNativeMessage() {
  message = "ipaddr"+document.getElementById('ipaddr').value;
  port.postMessage(message);
  message = "netmask"+document.getElementById('netmask').value;
  port.postMessage(message);
  message = "gateway"+document.getElementById('gateway').value;
  port.postMessage(message);
  message = "nameservers"+document.getElementById('nameservers').value;
  port.postMessage(message);
  if(document.getElementById('dhcp').checked==true)
  {
    message ='dhcp';
  }
  else
  {
    message ='staticip';
  }
  port.postMessage(message);
  message ='exec-command';
  port.postMessage(message);
  //appendMessage("Received message: <b>" + JSON.stringify(message) + "</b>");
}

function proxysetup() {
  if(document.getElementById('proxydirect').checked==true)
  {
    message = "proxytypedirect";
    port.postMessage(message);
    message ='exec-proxy';
    port.postMessage(message);
    document.getElementById('proxyserver').value="";
    return;
  }
  proxyserver = document.getElementById('proxyserver').value;
  //if(document.getElementById('proxyauto').checked==true)
  //{
    //message = "proxytypeauto";
    //port.postMessage(message);
    //message = "proxyserver" +  proxyserver ;
    //port.postMessage(message);
    //message ="exec-proxy";
    //port.postMessage(message);
    //return;
  //}
  if(document.getElementById('proxymanual').checked==true)
  {
    message = "proxytypemanual";
    port.postMessage(message);
    message = "proxyserver" +  proxyserver ;
    port.postMessage(message);
    message ='exec-proxy';
    port.postMessage(message);
    return;
  }
  //appendMessage("Received message: <b>" + JSON.stringify(message) + "</b>");
}

function onNativeMessage(message) {
  var inputstr=message.replace('"','');
  inputstr=inputstr.replace('"','');
  var stroffset=inputstr.search("ipaddr");
  if(stroffset>=0)
  {
    inputstr=inputstr.replace('ipaddr','');
    document.getElementById('ipaddr').value=inputstr;
    return;
  }
  var stroffset=inputstr.search("netmask");
  if(stroffset>=0)
  {
    inputstr=inputstr.replace('netmask','');
    document.getElementById('netmask').value=inputstr;
    return;
  }
  var stroffset=inputstr.search("gateway");
  if(stroffset>=0)
  {
    inputstr=inputstr.replace('gateway','');
    document.getElementById('gateway').value=inputstr;
    return;
  }
  var stroffset=inputstr.search("nameservers");
  if(stroffset>=0)
  {
    inputstr=inputstr.replace('nameservers','');
    inputstr=inputstr.replace(']','');
    document.getElementById('nameservers').value=inputstr;
    return;
  }
  var stroffset=inputstr.search("staticip");
  if(stroffset>=0)
  {
    document.getElementById('manual').checked = true;
    document.getElementById('dhcp').checked = false;
    return;
  }
  var stroffset=inputstr.search("dhcp");
  if(stroffset>=0)
  {
    document.getElementById('dhcp').checked = true;
    document.getElementById('manual').checked = false;
    return;
  }
  var stroffset=inputstr.search("proxytypedirect");
  if(stroffset>=0)
  {
    proxydirect();
    return;
  }
  //var stroffset=inputstr.search("proxytypeauto");
  //if(stroffset>=0)
  //{
    //proxyauto();
    //return;
  //}
  var stroffset=inputstr.search("proxytypemanual");
  if(stroffset>=0)
  {
    proxymanual()
    return;
  }
  var stroffset=inputstr.search("proxyserver");
  if(stroffset>=0)
  {
    inputstr=inputstr.replace('proxyserver','');
    inputstr=inputstr.replace(']','');
    document.getElementById('proxyserver').value=inputstr;
    return;
  }
  appendMessage("Received message: <b>" + JSON.stringify(inputstr) + "</b>");
}

function onDisconnected() {
  //appendMessage("Failed to connect: " + chrome.runtime.lastError.message);
  appendMessage("Failed to connect!!");
  port = null;
  updateUiState();
}

function connect() {
  var hostName = "network";
  appendMessage("Connecting to native messaging host <b>" + hostName + ".json" + "</b>")
  port = browser.runtime.connectNative(hostName);
  port.onMessage.addListener(onNativeMessage);
  port.onDisconnect.addListener(onDisconnected);
  updateUiState();
}

function proxydirect() {
  document.getElementById('proxydirect').checked = true;
  //document.getElementById('proxyauto').checked = false;
  document.getElementById('proxymanual').checked = false;
  updateUiState();
}

//function proxyauto() {
  //document.getElementById('proxydirect').checked = false;
  //document.getElementById('proxyauto').checked = true;
  //document.getElementById('proxymanual').checked = false;
  //updateUiState();
//}

function proxymanual() {
  document.getElementById('proxydirect').checked = false;
  //document.getElementById('proxyauto').checked = false;
  document.getElementById('proxymanual').checked = true;
  updateUiState();
}

document.addEventListener('DOMContentLoaded', function () {
  document.getElementById('connect-button').addEventListener('click', connect);
  document.getElementById('send-message-button').addEventListener('click', sendNativeMessage);
  document.getElementById('manual').addEventListener('click', manual);
  document.getElementById('dhcp').addEventListener('click', dhcp);
  document.getElementById('proxydirect').addEventListener('click', proxydirect);
  //document.getElementById('proxyauto').addEventListener('click', proxyauto);
  document.getElementById('proxymanual').addEventListener('click', proxymanual);
  document.getElementById('proxysetup').addEventListener('click', proxysetup);
  connect();
  updateUiState();
});

File: rootfs-4.9.88-aplex/home/user/native-messaging-python/network.py
#!/usr/bin/env python
# Copyright (c) 2012 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# A simple native messaging host. Shows a Tkinter dialog with incoming messages
# that also allows to send message back to the webapp.

import struct
import sys
import threading
import Queue
import os
import time

# Helper function that sends a message to the webapp.
def send_message(message):
   # Write message size.
  sys.stdout.write(struct.pack('I', len(message)))
  # Write the message itself.
  sys.stdout.write(message)
  sys.stdout.flush()

# Thread that reads messages from the webapp.
def read_thread_func(queue):
  message_number = 0
  output = os.popen("connmanctl services | awk '{print $3}'")
  devicename = str(output.read()).rstrip()
  if devicename:
    command = "connmanctl services " + str(devicename) +" | grep 'IPv4 =' | grep Method=manual"
    output = os.popen(command)
    ethernettype = str(output.read()).rstrip()
    if ethernettype:
      send_message('"staticip"')
    else:
      send_message('"dhcp"')
    output = os.popen("/sbin/ifconfig 'eth0' | grep 'inet addr:' | awk '{print $2}' | awk -F ':' '{print $2}'")
    ipaddr = str(output.read()).rstrip()
    if ipaddr :
      sendtext='"'+ "ipaddr" +str(ipaddr )+'"'
      send_message(sendtext)
    output = os.popen("/sbin/ifconfig 'eth0' | grep 'inet ' | awk  '{print $4}' | awk -F ':' '{print $2}'")
    netmask = str(output.read()).rstrip()
    if netmask :
      sendtext='"'+ "netmask" +str(netmask)+'"'
      send_message(sendtext)
    output = os.popen("/sbin/route -n | grep UG | head -n  1 | awk '{print $2}'")
    gateway = str(output.read()).rstrip()
    if gateway:
      sendtext='"'+ "gateway" +str(gateway)+'"'
      send_message(sendtext)
    if devicename :
      command = "connmanctl services " + str(devicename) +" | grep 'Nameservers =' | awk '{print $4}'"
      output = os.popen(command)
      nameservers = str(output.read()).rstrip()
      #nameservers = nameservers.replace(']','')
      if nameservers :
        sendtext='"'+ "nameservers" +str(nameservers )+'"'
        send_message(sendtext)
      command = "connmanctl services " + str(devicename) +" | grep 'Proxy ='"
      output = os.popen(command)
      proxytext = str(output.read()).rstrip()
      findret = proxytext.find('direct')
      if findret > -1 :
        teststr='"' + "proxytypedirect" + '"'
        send_message(teststr)
      findret = proxytext.find('auto')
      if findret > -1 :
        teststr='"' + "proxytypeauto" + '"'
        send_message(teststr)
        strstart= proxytext.find('URL=')
        if strstart > -1 :
          strstart=strstart+4
          strend = proxytext.find(',',strstart)
          serverstr = proxytext[strstart:strend]
          teststr='"' + "proxyserver" + serverstr + '"'
          send_message(teststr)
      findret = proxytext.find('manual')
      if findret > -1 :
        teststr='"' + "proxytypemanual" + '"'
        send_message(teststr)
        strstart = proxytext.find('Servers=[ ')
        if findret > -1 :
          strstart= strstart + 10
          strend = proxytext.find(' ]',strstart)
          serverstr = proxytext[strstart:strend]
          teststr='"' + "proxyserver" + serverstr + '"'
          send_message(teststr)
  while 1:
    # Read the message length (first 4 bytes).
    text_length_bytes = sys.stdin.read(4)
    if len(text_length_bytes) == 0:
      if queue:
        queue.put(None)
      sys.exit(0)

    # Unpack message length as 4 byte integer.
    text_length = struct.unpack('i', text_length_bytes)[0]

    # Read the text (JSON object) of the message.
    text = sys.stdin.read(text_length);
    textstr =text.strip('"')
    if queue:
      queue.put(text)
    else:
      # In headless mode just send an echo message back.
      findret = textstr.find('ipaddr')
      if findret > -1 :
        ipaddr = textstr.strip('ipaddr')
        #teststr='"' + str(ipaddr) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('netmask')
      if findret > -1 :
        netmask= textstr.strip('netmask')
        #teststr='"' + str(netmask) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('gateway')
      if findret > -1 :
        gateway= textstr.strip('gateway')
        #teststr='"' + str(gateway) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('nameservers')
      if findret > -1 :
        nameservers= textstr.strip('nameservers')
        #teststr='"' + str(nameservers) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('dhcp')
      if findret > -1 :
        ethernettype= 'dhcp'
        continue
      findret = textstr.find('staticip')
      if findret > -1 :
        ethernettype= 'manual'
        continue
      findret = textstr.find('proxytype')
      if findret > -1 :
        proxytype= textstr.replace('proxytype','')
        #teststr='"' + str(proxytype) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('proxyserver')
      if findret > -1 :
        proxyserver= textstr.replace('proxyserver','')
        #teststr='"' + str(proxyserver) + '"'
        #send_message(teststr)
        continue
      findret = textstr.find('exec-proxy')
      if findret > -1 :
        if proxytype == 'direct':
          command = "connmanctl config " + str(devicename) +" proxy direct "
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          command = "echo \"\" > /etc/proxyserver"
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          output = os.popen("sync")
          proxyret =str(output.read()).rstrip()
        if proxytype == 'auto':
          command = "connmanctl config " + str(devicename) +" proxy auto " + str(proxyserver)
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          command = "echo " + str(proxyserver) + " > /etc/proxyserver"
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          output = os.popen("sync")
          proxyret =str(output.read()).rstrip()
        if proxytype == 'manual':
          command = "connmanctl config " + str(devicename) +" proxy manual " + str(proxyserver)
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          command = "echo " + str(proxyserver) + " > /etc/proxyserver"
          output = os.popen(command)
          proxyret =str(output.read()).rstrip()
          output = os.popen("sync")
          proxyret =str(output.read()).rstrip()
        continue
      findret = textstr.find('exec-command')
      if findret > -1 :
        if ethernettype== 'manual':
          command = "connmanctl config " + str(devicename) +" ipv4 manual "+ str(ipaddr) +" "+str(netmask) +" "+str(gateway)
          output = os.popen(command)
          ethernetret = str(output.read()).rstrip()
          command = "connmanctl config " + str(devicename) +" nameservers "+ str(nameservers)
          output = os.popen(command)
          ethernetret = str(output.read()).rstrip()
        else:
          command = "connmanctl config " + str(devicename) +" ipv4 dhcp"
          output = os.popen(command)
          ethernetret = str(output.read()).rstrip()
          command = "connmanctl config " + str(devicename) +" nameservers "+ str(nameservers)
          output = os.popen(command)
          ethernetret = str(output.read()).rstrip()
        output = os.popen("sync")
        devicename = output.read()
        time.sleep(3)
        output = os.popen("connmanctl services | awk '{print $3}'")
        devicename = str(output.read()).rstrip()
        if devicename:
          command = "connmanctl services " + str(devicename) +" | grep 'IPv4 =' | grep Method=manual"
          output = os.popen(command)
          ethernettype = str(output.read()).rstrip()
          if ethernettype:
            send_message('"staticip"')
          else:
            send_message('"dhcp"')
          output = os.popen("/sbin/ifconfig 'eth0' | grep 'inet addr:' | awk '{print $2}' | awk -F ':' '{print $2}'")
          ipaddr = str(output.read()).rstrip()
          if ipaddr :
            sendtext='"'+ "ipaddr" +str(ipaddr )+'"'
            send_message(sendtext)
          output = os.popen("/sbin/ifconfig 'eth0' | grep 'inet ' | awk  '{print $4}' | awk -F ':' '{print $2}'")
          netmask = str(output.read()).rstrip()
          if netmask :
            sendtext='"'+ "netmask" +str(netmask)+'"'
            send_message(sendtext)
          output = os.popen("/sbin/route -n | grep UG | head -n  1 | awk '{print $2}'")
          gateway = str(output.read()).rstrip()
          if gateway:
            sendtext='"'+ "gateway" +str(gateway)+'"'
            send_message(sendtext)
          if devicename :
            command = "connmanctl services " + str(devicename) +" | grep 'Nameservers =' | awk '{print $4}'"
            output = os.popen(command)
            nameservers = str(output.read()).rstrip()
            #nameservers = nameservers.replace(']','')
            if nameservers :
              sendtext='"'+ "nameservers" +str(nameservers )+'"'
              send_message(sendtext)
 

def Main():
    read_thread_func(None)
    sys.exit(0)

if __name__ == '__main__':
  Main()
